internal/dag/conditions.go

```diff
@@ -15,17 +15,34 @@ package dag
 
 import (
 	"fmt"
-	"path"
+	"regexp"
 	"strings"
 
 	projcontour "github.com/projectcontour/contour/apis/projectcontour/v1"
 )
 
-func pathCondition(conds []projcontour.Condition) Condition {
-	prefix := "/"
+// mergePathConditions merges the given slice of prefix Conditions into a single
+// prefix Condition.
+// pathConditionsValid guarantees that if a prefix is present, it will start with a
+// / character, so we can simply concatenate.
+func mergePathConditions(conds []projcontour.Condition) Condition {
+	prefix := ""
```

Comment:
what about
```
prefix := "/"
trailingSlash := false
for _, cond := range cones {
    cp := cond.Prefix
    if cp != "" {
          trailingSlash = cp[len(cp)-1] == '/'
    }
    prefix = path.Join(prefix, cp)
}
if trailingSlash {
   prefix += "/"
}
```

Reply:
Sure, that seems reasonable. Certainly saves on a regexp usage, which seems nice.

Reply:
Actually, turns out that path.Join does not trim repeated `/` characters, so I'm going to stick with the current string concatenation implementation.